### ISSUE TITLE
Make code coverage work again

### ISSUE THREAD

**Issue by jtattermusch** [grpc/grpc#8824]
Currently all the code coverage reports are timing out.
Before that, code coverage in some languages languages wasn't always available.

This is not urgent, but at some point, we need to revisit this and make sure we are getting results reliably. Perhaps we could use some of the new infrastructure tricks that we've introduced in the past month.

https://grpc-testing.appspot.com/job/gRPC_coverage/8838/
https://github.com/grpc/grpc/issues/6672

**Comment by matt-kwong** [comment]
A big issue was that builds were failing because of a bad worker, but that worker is now disabled. I upped the timeout time from Jenkins, and from the completed build I see these reasons for builds timing out:

- We added a lot of tests (over 2000 more compared to the last successful build in June)
- More tests are taking longer, which can be seen by sorting by duration for these two builds: https://grpc-testing.appspot.com/job/gRPC_coverage/9081/testReport/(root)/tests/ and https://grpc-testing.appspot.com/job/gRPC_coverage/lastStableBuild/testReport/(root)/tests/
- I believe that the time it takes for tests to time out have been increased over the last few months

I'm not too sure what can be done about this, but at least builds are completing and after more builds complete, we'll see if the results are reliable.

**Comment by matt-kwong** [comment]
C# tests are failing when all languages are being ran, but they pass when running only C# tests. See https://grpc-testing.appspot.com/job/gRPC_coverage/9122/testReport/(root)/tests/ vs https://grpc-testing.appspot.com/job/gRPC_coverage/9120/testReport/. So, it seems like something in the previous tests are breaking something. I'll look into it, but I just wanted to send an update in case this might have happened before and you know what the issue is.

**Comment by jtattermusch** [comment]
The problems seems to be loading of C# native library:

```
System.IO.IOException : Error loading native library "/var/local/git/grpc/src/csharp/Grpc.Reflection.Tests/bin/Debug/libgrpc_csharp_ext.x64.so"
TearDown : System.NullReferenceException : Object reference not set to an instance of an object
  at Grpc.Core.Internal.UnmanagedLibrary..ctor (System.String[] libraryPathAlternatives) [0x00068] in <8add62ce508f4dcbbd89255d51bd80c4>:0 
  at Grpc.Core.Internal.NativeExtension.Load () [0x0005d] in <8add62ce508f4dcbbd89255d51bd80c4>:0 
  at Grpc.Core.Internal.NativeExtension..ctor () [0x00007] in <8add62ce508f4dcbbd89255d51bd80c4>:0 
  at Grpc.Core.Internal.NativeExtension.Get () [0x0002c] in <8add62ce508f4dcbbd89255d51bd80c4>:0 
  at Grpc.Core.Internal.NativeMethods.Get () [0x00001] in <8add62ce508f4dcbbd89255d51bd80c4>:0 
  at Grpc.Core.GrpcEnvironment.GrpcNativeInit () [0x00001] in <8add62ce508f4dcbbd89255d51bd80c4>:0 
  at Grpc.Core.GrpcEnvironment..ctor () [0x0001f] in <8add62ce508f4dcbbd89255d51bd80c4>:0 
  at Grpc.Core.GrpcEnvironment.AddRef () [0x0002e] in <8add62ce508f4dcbbd89255d51bd80c4>:0 
  at Grpc.Core.Server..ctor (System.Collections.Generic.IEnumerable`1[T] options) [0x0006e] in <8add62ce508f4dcbbd89255d51bd80c4>:0 
  at Grpc.Core.Server..ctor () [0x00000] in <8add62ce508f4dcbbd89255d51bd80c4>:0 
  at Grpc.Reflection.Tests.ReflectionClientServerTest.Init () [0x0001a] in <a645daf926c34570a66c4b6afe98cc4b>:0 
  at (wrapper managed-to-native) System.Reflection.MonoMethod:InternalInvoke (System.Reflection.MonoMethod,object,object[],System.Exception&)
  at System.Reflection.MonoMethod.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00038] in <94fd79a3b7144c54b4cb162b50fc7761>:0 
```

There's two possible reasons:
1. the library is not in the given location (it should be copied to there by msbuild when building the C# solution)
2. the library cannot be loaded because there are some symbols missing (C# is built with make EMBED_OPENSSL=true option, while C core isn't).

It seems that it might be 2), because if ran alone (= C core does not get built), things work just fine.

We've actually seen a similar issue with performance tests:
https://github.com/grpc/grpc/blob/66de9494afd6aa5eaddb5f8fe66f275c4b2a8e38/tools/run_tests/performance/build_performance.sh#L43

I'll try think of a workaround to use here...

**Comment by matt-kwong** [comment]
It looks like the same issue was supposed to be addressed by #7401. However, the last successful build was https://grpc-testing.appspot.com/job/gRPC_coverage/6482/ which predates when that PR was merged. Also, the builds prior to and after 6482 have the same issue with C#. Also they're built on the same machine and tested the same revision of gRPC, so I'm not sure what to make of this flaky successful build.

**Comment by matt-kwong** [comment]
https://grpc-testing.appspot.com/job/gRPC_coverage/9200/ - Here's another flake that was the only one to run C# tests successfully in the last few thousand builds. I tested for a possible race condition by running `inner_jobs=1` on a fresh VM, but that didn't work so I'm still not sure why this build was successful. Also, I talked with Nico, and he said that setting `EMBED_OPENSSL=true` for C core tests should be fine, and I have had positive results doing this, so I think it should be fine to do this just for the coverage test.
